pkg/config/config.go

```diff
@@ -37,37 +37,64 @@ var (
 	validRoles = []string{"controlplane", "node"}
 )
 
-type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+// Top level config
+type Config struct {
+	Core Core `json:core`
 }
 
-type DebugConfig struct {
-	Pprof bool `json:"pprof"`
+// Core config
+type Core struct {
+	Network   Network   `json:network`
+	DNS       DNS       `json:dns`
+	Debugging Debugging `json:debugging`
```

Comment:
At the risk of entering bikeshed territory... is this a cross-cutting concern? On OCP, flags for things like log verbosity live within the operator/v1 group because they relate to the operation of the cluster components and not their functional behavior. When we discussed configuration a couple weeks ago, I stuck log level outside of core in the doc example because I thought there was consensus.

Reply:
I'd be happy with a log level option at the top level if that's what the team agrees on.